### ISSUE TITLE
Follow.OfCurrentUser breaks with No type given exception

### ISSUE THREAD

**Issue by YettySpaghetty** [JohnnyCrazy/SpotifyAPI-NET#461]
When trying to call var following = await spotify.Follow.OfCurrentUser(), it breaks with SpotifyAPI.Web.APIException: 'No type given'.

I assume it's because not type (either User or Artist) was provided for the request. I can't seem to find how pass the type anywhere although the OfCurrentUser() method does take one overload.

**Comment by JohnnyCrazy** [comment]
I see, there is definitely a bug here, but it should not occur with `OfCurrentUser()`.

So this example works for me:

```csharp
var following = await spotify.Follow.OfCurrentUser();
Console.WriteLine(following.Artists);
```

while the following throws `No type given`, the request params are ignored:

```csharp
var following = await spotify.Follow.OfCurrentUser(
  new FollowOfCurrentUserRequest(FollowOfCurrentUserRequest.Type.Artist)
);
Console.WriteLine(following.Artists);
```

Can you confirm this behavior? beta.4 will contain a fix for the second part.

**Comment by YettySpaghetty** [comment]
![image](https://user-images.githubusercontent.com/38431233/84505607-b21d7b00-acc6-11ea-995f-900da64b2f97.png)

Unfortunately none of the examples work for me, but I do believe it's not from the OfCurrentUser() method when I look at the stack trace.

**Comment by JohnnyCrazy** [comment]
Sadly you cut of the first part of the stacktrace, are you 100% calling `OfCurrentUser()` without any parameters? Would need to see some more code and all of the stacktrace to further drill down the problem.

**Comment by YettySpaghetty** [comment]
![image](https://user-images.githubusercontent.com/38431233/84506500-2278cc00-acc8-11ea-9e05-a596f98d1eee.png)


SpotifyAPI.Web
   at SpotifyAPI.Web.Http.APIConnector.ProcessErrors(IResponse response)
   at SpotifyAPI.Web.Http.APIConnector.<DoRequest>d__31.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at SpotifyAPI.Web.Http.APIConnector.<DoSerializedRequest>d__30`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at SpotifyAPI.Web.Http.APIConnector.<SendAPIRequest>d__34`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at SpotifyServer.SpotifyAPI.<GetPrivateUserFriends>d__6.MoveNext() in C:\Users\talpe\source\repos\Spotify Social\SpotifyServer\SpotifyAPI.cs:line 73
No type given

This is the full error when running the code above. The authorization works, as I can get the my Private Profile data no problem and I did provide the necessary scopes for UserFollowRead

**Comment by JohnnyCrazy** [comment]
Ah, one try, which version are you on? beta.3 contains an import bug fix for query parameters.

**Comment by YettySpaghetty** [comment]
![image](https://user-images.githubusercontent.com/38431233/84508948-bdbf7080-accb-11ea-8e87-160814d62fc1.png)
Beta.3 yes. I'll try reinstalling the package in case I got some corrupted files

**Comment by YettySpaghetty** [comment]
When I query the e.Data in the catch block I receive a System.Collections.ListDictionaryInternal message. Maybe it helps

**Comment by JohnnyCrazy** [comment]
Okay, I'm out of clues right now.

I created a new .NET Standard project with the following code and still can't reproduce your errors

```
var config = SpotifyClientConfig
    .CreateDefault()
    .WithHTTPLogger(new SimpleConsoleHTTPLogger())
    .WithToken("MyToken");
var spotify = new SpotifyClient(config);

var test2 = await spotify.UserProfile.Current();
var test = await spotify.Tracks.Get("65jw6GhwQW3Db55PxlPEpS", new TrackRequest { Market = "DE" });
var following = await spotify.Follow.OfCurrentUser();
Console.WriteLine(following.Artists);
```

Could you try to also add `.WithHTTPLogger(new SimpleConsoleHTTPLogger())` to your setup and see if it correctly contains the request parameter or not? And also, what .NET Framework and OS (I assume Windows 10) are you on?

**Comment by YettySpaghetty** [comment]
`GET me []
--> OK application/json {  "country" : "RO",  "display_name" : "Alexandru-


GET tracks/65jw6GhwQW3Db55PxlPEpS [market=DE]
--> OK application/json {  "album" : {    "album_type" : "album",    "arti


GET me/following [type=artist]
--> BadRequest application/json {  "error" : {    "status" : 400,    "message" : "

System.Collections.ListDictionaryInternal
SpotifyAPI.Web
   at SpotifyAPI.Web.Http.APIConnector.ProcessErrors(IResponse response)
   at SpotifyAPI.Web.Http.APIConnector.<DoRequest>d__31.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at SpotifyAPI.Web.Http.APIConnector.<DoSerializedRequest>d__30`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at SpotifyAPI.Web.Http.APIConnector.<SendAPIRequest>d__34`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at SpotifyServer.SpotifyAPI.<GetPrivateUserFriends>d__6.MoveNext() in C:\Users\talpe\source\repos\Spotify Social\SpotifyServer\SpotifyAPI.cs:line 78
No type given
` 
This is what I got, along with the stack trace. 

It's windows 10 yes. 4.6.1 .NET Framework

**Comment by JohnnyCrazy** [comment]
Thanks, I can reproduce it now, seems to be related to the .NET Framework Version. Will get back to you once I found a fix.

**Comment by JohnnyCrazy** [comment]
Query params are not applied correctly, there is a double leading `?` instead of a single one when using .NET Framework 4.6.1. See the example below where first 3 requests are via .NET Framework 4.6.1 and the other 3 are via .NET Standard

![](https://i.johnnycrazy.de/uzgNix_YOcF87pQPSRfUU)

**Comment by YettySpaghetty** [comment]
Should I roll back to an earlier version?

**Comment by JohnnyCrazy** [comment]
I should be able to release a fix within the next hours, I don't recommend going to 5.x

**Comment by JohnnyCrazy** [comment]
Fixed by #462 and released as beta.4

**Comment by YettySpaghetty** [comment]
Thanks a lot for the promptness! It works on my machine with the new package update
